formapp/static/js/home.js

```diff
@@ -4,6 +4,28 @@ let thumbButtons = document.querySelectorAll(".thumbButton");
 let letterButtons = document.querySelectorAll(".letterButton");
 let chosenThumb = false;
 let chosenLetter = false;
+let meRes;
+let bgmRes;
+let modeRes;
+let permsRes;
+let pixRes;
+
+async function getApiData(first) {
+  let apiData = await Promise.all([
+    getResponse("/getme"),
+    getResponse("/getbgm"),
+    getResponse("/getmode"),
+    getResponse("/getpermissions"),
+    getResponse("/getpix")
+  ]);
+  meRes = apiData[0];
+  bgmRes = apiData[1];
+  modeRes = apiData[2];
+  permsRes = apiData[3];
+  pixRes = apiData[4];
+  if (first) init();
+  else update();
+}
```

Comment:
You'll have to explain to me what this is doing.

Reply:
`getResponse` is defined in header.html. It parses and returns the response text from an endpoint.
This code waits for each response (using `Promise.all` to get them concurrently). Once the promise is fulfilled, the data from each URL is saved to a variable.
If this is the first time this function is executed, call `init()`; otherwise, call `update()`. Since this code is shared by basic.html and advaced.html, each page has its own `init` and `update` functions.